app/controllers/blackouts_controller.rb

```diff
@@ -93,10 +93,30 @@ def create # rubocop:disable MethodLength
   end
 
   def update
+    p = blackout_params
+    p[:created_by] = current_user.id 
     @blackout.set_id = nil
-    if @blackout.update_attributes(blackout_params)
-      redirect_to @blackout, notice: 'Blackout was successfully updated.'
+
+    # check for conflicts
+    res = Reservation.overlaps_with_date_range(p[:start_date], p[:end_date])
+          .active
+
+    # save and exit
+    if res.empty? && @blackout.update_attributes(p)
+      redirect_to @blackout, notice: 'Blackout was successfully updated.' 
     else
+      if res.any?
```

Comment:
I don't think we need this `if` block -- we want the detailed reservation info always.